Feature/test requirements (#20)
distributed-technologies/wiki: helm/test.md

# Testing Services

> All of the services develloped for the platform should pass basic acceptance tests before they are deployed. 

Reviewer: ```suggestion
All of the services developed for the platform should pass basic acceptance tests before they are deployed. 
```

> The tests should verify the core functionality of the service. The tests should catch configuration issues or problems with the service and ensure that the service functions as is inteded. 

Reviewer: ```suggestion
The tests should verify the core functionality of the service. The tests should catch configuration issues or problems with the service and ensure that the service functions as intended. 
```

> Generally the tests are located in a Helm proxy chart for the service. The tests are located in a folder called `test`. Before any merge of changes onto the main branch, a github workflow automatically runs the test in a `kind` cluster. 

Reviewer 2: For consistency, I think we should use the helm test hooks for acceptance tests.  
Then a test is described in yaml as a Pod that runs a test script, and we can use the `helm test xxx` command as a test engine. That way we can run tests inside the cluster and we are not dependent on the host system having packages installed.
See https://github.com/distributed-technologies/services-kafka/blob/main/.github/workflows/test_create.yaml
for reference.

Author: ```suggestion
Generally the tests are located in a Helm proxy chart for the service. The tests are located in a folder called `test`. Before any merge of changes onto the main branch, a github workflow automatically runs the test in a `kind` cluster. 
See https://github.com/distributed-technologies/services-kafka/blob/main/.github/workflows/test_create.yaml
for reference.
```
